digitiser-aggregator/src/frame/cache.rs

```diff
@@ -27,52 +24,84 @@ where
         }
     }
 
-    pub(crate) fn push(&mut self, digitiser_id: DigitizerId, metadata: &FrameMetadata, data: D) {
-        match self
-            .frames
-            .iter_mut()
-            .find(|frame| frame.metadata.equals_ignoring_veto_flags(metadata))
-        {
-            Some(frame) => {
-                frame.push(digitiser_id, data);
-                frame.push_veto_flags(metadata.veto_flags)
-            }
-            None => {
-                let mut frame = PartialFrame::<D>::new(self.ttl, metadata.clone());
-                frame.push(digitiser_id, data);
-                self.frames.push_back(frame);
+    #[tracing::instrument(skip_all, fields(
+        digitiser_id = digitiser_id,
+        metadata_timestamp = metadata.timestamp.to_rfc3339(),
+        metadata_frame_number = metadata.frame_number,
+        metadata_period_number = metadata.period_number,
+        metadata_veto_flags = metadata.veto_flags,
+        metadata_protons_per_pulse = metadata.protons_per_pulse,
+        metadata_running = metadata.running
+    ))]
+    pub(crate) fn push<'a>(
+        &'a mut self,
+        digitiser_id: DigitizerId,
+        metadata: &FrameMetadata,
+        data: D,
+    ) {
+        let frame = {
+            match self
+                .frames
+                .iter_mut()
+                .find(|frame| frame.metadata.equals_ignoring_veto_flags(metadata))
+            {
+                Some(frame) => {
+                    frame.push(digitiser_id, data);
+                    frame.push_veto_flags(metadata.veto_flags);
+                    frame
+                }
+                None => {
+                    let mut frame = PartialFrame::<D>::new(self.ttl, metadata.clone());
+
+                    // Initialise the span field
+                    if let Err(e) = frame.span_init() {
+                        warn!("Frame span initiation failed {e}")
+                    }
+
+                    frame.push(digitiser_id, data);
+                    self.frames.push_back(frame);
+                    self.frames.back().expect("Partial Frame Exists")
```

Comment:
`expect()` strings should contain the word "should", the panic message this generates would be ambiguous about if the error is the existence or absence of a partial frame.

I have also been adding ", this should never fail" to panic messages that are by (and only by) guarantees of the APIs used and the way they are used never going to fail.

Reply:
I did not mean entirely replace the message with "this should never fail", the format should be "\<assertion including the word should\>, this should never fail".

Reply:
Sorry, got ya now.